### ISSUE TITLE
Three potential inaccurate issues to be confirmed

### ISSUE THREAD

**Issue by liangjinsong** [qichao1984/NCyc#7]
Hi! While the using of NCycDB, I found some potential inaccurate issues in it. So I want to confirm them here from the author for further improvement of this useful database.
1.  441 gene names (and sequence) occur twice in the file NCyc_100.faa. I think the duplicated gene sequences should be removed in the file. For your information, ten duplicated gene names are listed below:
1000565.METUNv1_00600
100226.SCO5525
1002339.HMPREF9373_0600
1002339.HMPREF9373_0712
1003195.SCAT_4357
1004836.SXCC_03526
1006006.Mcup_0918
1006551.KOX_23300
1006551.KOX_27050
1026882.MAMP_00572

2. 81 gene names occur twice in the file id2gene.map, and what's more, the classification of any of these gene names are not exactly the same. Ten of the mentioned genes are listed below for your information.
A0A060I505      napA
A0A060I505      nasA
A0A0D6FVV7      narG
A0A0D6FVV7      narZ
A0A127KYU8      nasB
A0A127KYU8      nirB
A0A142WVR2      narB
A0A142WVR2      nasA
A0A142Y6U0      narB
A0A142Y6U0      nasA
A0A142YH98      narB
A0A142YH98      nasA
A0A145CM87      narB
A0A145CM87      nasA
A0A1D8AZ75      narB
A0A1D8AZ75      nasA

3. The gene names should be identical between the two files NCyc_100.faa and id2gene.map. But 54410 gene names in file NCyc_100.faa can not found in file id2gene.map, and 99288  gene names in file id2gene.map can not found in file NCyc_100.faa. The issue will cause inaccuracy when calculating the relative abundance of Nitrogen cycling genes (sub)families.

**Comment by ajray34** [comment]
I also experienced issues with IDs not present in id2gene.map.

**Comment by qichao1984** [comment]
Dear users, thanks for pointing out potential issues in this database! I will look into issue #1 and #2 pointed out by Liang asap.
For issue #3, this was actually intended in order to reduce false positive assignments in Ncyc profiling, i.e. the IDs not showing up in id2gene.map belong to Ncyc gene homologs, not Ncyc gene families. By doing so, reads having better "blast" hit scores with Ncyc homologs will not show up in the final Ncyc profile.


________________________________

**Comment by rprops** [comment]
Any update on this issue? This could have implications for an analysis I'm currently running? 
Thank you

**Comment by rprops** [comment]
Just checking if there has been any update? I've noticed some database changes but none that were incorporated into the script.

**Comment by qichao1984** [comment]
Yes, changes have been made to the database. However, nothing need to be done for the script. Please feel free to use the updated database! Thanks for figuring out these issues!

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by rprops** [comment]
Ok just checking because the perl script still has this chunk refering to the old `id2gene.map` file instead of `id2gene.map.2019Jul`:
```
my %id2gene;
open( FILE, "data/id2gene.map" ) || die "#1\n";
while (<FILE>) {
  chomp;
  my @items = split( "\t", $_ );
  $id2gene{ $items[0] } = $items[1];
}
close FILE;
```
